third_party/amd/backend/compiler.py

```diff
@@ -290,6 +291,8 @@ def make_llir(src, metadata, options):
         # LDS size is determined by provided arch name.
         custom_lds_size = 0
         amd.passes.ttgpuir.add_optimize_lds_usage(pm, options.arch, custom_lds_size)
+        amd.passes.ttgpuir.add_warp_pipeline_conversion(pm)
+        passes.common.add_canonicalizer(pm)
```

Comment:
do we really need another full canonicalization pass here? Might be better to do targeted cleanups